tools/hacked/splitFasta/splitFasta.py

```diff
@@ -0,0 +1,14 @@
+import sys, os
+if __name__ == "__main__":
+    #assuming perfect input, read every two lines
+    inpath = sys.argv[1]
+    file_contents = open(inpath, 'r').readlines()
+    os.makedirs('splits')
+    inname = os.path.basename(inpath)
+    for i in range(0, len(file_contents), 2):
+        headline = file_contents[i]
+        outname = headline[1:headline.index(' ')]+'.fa'
+        outfile = open(os.path.join('splits',outname), 'w')
+        outfile.write(file_contents[i])
+        outfile.write(file_contents[i+1])
+        outfile.close()
```

Comment:
The script breaks on non-perfect input... I had a user who ran this and now has >2k items in their history because of fixed width fasta format.

So, I rewrote this a little, hope it is useful.

```python
50.24.16.65 - esr [16/Oct/2015:14:19:06 -0500] "GET /galaxy/api/histories/e94e536920e6936f/contents HTTP/1.1" 200 - "https://cpt.tamu.edu/galaxy/root" "Mozillimport sys                                                                                                                                                    
from Bio import SeqIO                                                                                                                                         
                                                                                                                                                              
if __name__ == "__main__":                                                                                                                                    
    inpath = sys.argv[1]                                                                                                                                      
                                                                                                                                                              
    with open(inpath, 'r') as handle:                                                                                                                         
        for record in SeqIO.parse(handle, 'fasta'):                                                                                                           
            header = record.id + '.fa'                                                                                                                        
            with open(header, 'w') as handle2:                                                                                                                
                SeqIO.write([record], handle2, 'fasta')                                                                                                       
```